connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -123,4 +131,14 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
       )
     }
   }
+
+  test("SPARK-49162: Push down date_trunc function") {
```

Comment:
This is good! Could we write a few more tests, maybe test some edge cases. We are adding a new pushdown, let's make sure there are no correctness issues.

Reply:
Also, would be good to have some other variable in trunc function other than 'DAY'